internal/hmyapi/apiv1/types.go

```diff
@@ -157,12 +157,15 @@ func newRPCCXReceipt(cx *types.CXReceipt, blockHash common.Hash, blockNumber uin
 
 // newRPCTransaction returns a transaction that will serialize to the RPC
 // representation, with the given location metadata set (if available).
-func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber uint64, timestamp uint64, index uint64) *RPCTransaction {
-	var signer types.Signer = types.FrontierSigner{}
-	if tx.Protected() {
-		signer = types.NewEIP155Signer(tx.ChainID())
+// Note that all txs on Harmony are replay protected (post EIP155 epoch).
+func newRPCTransaction(
+	tx *types.Transaction, blockHash common.Hash,
+	blockNumber uint64, timestamp uint64, index uint64,
+) *RPCTransaction {
+	from, err := types.PoolTransactionSender(types.NewEIP155Signer(tx.ChainID()), tx)
```

Comment:
will this break for before EIP155? why can't you keep the previous logic to choose different singer?

Reply:
We are never in non EIP155 epoch so it will never break, confirmed with @rlan35